### PR TITLE
bootstrap project

### PR DESCRIPTION
**Pull request by BusyJay**
- Add serveral important traits
- Fix license and README with updated name

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+///   handle -> pause
+///     |
+///    end
+/// ```
+/// 
+/// Generally users should use the provided future thread pool or callback
+/// thread pool instead. This is only for advance customization. 
+pub trait Runner {
+    /// The task runner can handle.
+    type Task;
```
Comment (sticnarf) [inline src/lib.rs]: I once met some trouble if `Task` knows nothing about the queue (`LocalSpawn` and `RemoteSpawn` here).

For example, a callback handle should be aware of what `LocalSpawn` is, but with this trait it's hard to do that.

**Review by sticnarf — APPROVED** [review]
LGTM

File: src/runner.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+pub trait Runner {
+    /// The task runner can handle.
+    type Task;
+
+    /// The local spawn that can be accepted to spawn tasks.
+    type Spawn: LocalSpawn<Task = Self::Task>;
+
+    /// Called when the runner is started.
+    ///
+    /// It's guaranteed to be the first method to called before anything else.
```
Comment (AndreMouche) [inline src/runner.rs]: to call or to be called?

**Review by AndreMouche — COMMENTED** [review]
Rest LGTM

File: src/runner.rs (hunk 2 above)
Comment (BusyJay) [reply src/runner.rs]: My mistake.

**Review by AndreMouche — APPROVED** [review]
LGTM

**Pull request merged**